recirq/seniority_zero/circuits_expt/core.py

```diff
@@ -0,0 +1,430 @@
+# Copyright 2023 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Implementation of the actual circuits used in the s0 experiment
+
+To ensure precise scheduling + echo insertion, much of this is hard-coded and
+long-winded. We recommend using circuits_simplified to understand what is going
+on in the experiment.
+"""
+from typing import List, Optional
+
+import cirq
+
+from recirq.seniority_zero.circuits_expt.gates_this_experiment import cmnot, cnot, gsgate, swap
+from recirq.seniority_zero.misc import (
+    parallelize_circuits,
+    safe_concatenate_circuits,
+    safe_tetris_circuits,
+)
+from recirq.seniority_zero.scheduling import get_tqbg_groups
+
+
+def _make_lightcone(num_qubits, depth=None):
+    """Make list of gates which are hit by a given qubit's backwards light cone.
+    Assumes a brickwall ansatz
```

Comment:
here and elsewhere: please ensure that the docstrings start with a one line summary optionally followed by a blank line and the rest of the docs.

This format is important for the docs system. The summary line must be one line. You can use 100 character line length

Reply:
I think I've gotten all of these.